### DISCUSSION TITLE
Where can I learn more about what happens under the hood when you redeploy a Dagster deployment with new code? What happens to in-progress jobs?

### DISCUSSION THREAD

**Discussion by dagsir** [dagster-io/dagster#18054] [Q&A]
Where can I learn more about what happens under the hood when you redeploy a Dagster deployment with new code? What happens to in-progress jobs?

---

*The question was originally asked in [Dagster Slack](https://dagster.slack.com/archives/C01U954MEER/p1700091830235949?thread_ts=1700091830.235949&cid=C01U954MEER).*

**Comment by prha** [comment]
It depends on what run launcher / executor you are using.  

The default run launcher executes job runs as a process on the code sever.  These runs are affected by code location deployments.  By default, these runs spawn a new process for every op/step (using the multiprocess executor), and so ops executed after the deployment will be running updated code.  This [overview](https://docs.dagster.io/deployment/overview#open-source-deployment-architecture) doesn't address redeployment specifically, but is useful at mapping out the different abstractions.

The other Dagster-maintained run launchers (k8s, ecs, docker) launch runs in isolation from the code servers, so they are unaffected by redeployments.  The run will continuing executing the code as defined at the time of the run launch.  This is true for both serverless and hybrid deployments in Dagster Cloud.

**Comment by skeller88** [comment]
Just to make sure I'm understanding [this diagram](https://docs.dagster.io/dagster-cloud/deployment/hybrid), each of the workers/executors in our cloud have a dagster API client installed that enables them to stream job metadata back to Dagster Cloud. This means that we can redeploy the agent while jobs are running, because then the agent will be able to load the state of jobs from Dagster Cloud once it redeploys?
